### ISSUE TITLE
apr serve --gpu batched decode generates all token_id=0 (broken output)

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#659]
## Description

When `apr serve run --gpu` processes concurrent requests, the batched decode step produces token_id=0 for every position in every batch:

```
[PMAT-072] decode step 0: token_ids=[0, 0, 0, 0, 0], done=[false, false, false, false, false]
[PMAT-072] decode step 1: token_ids=[0, 0, 0, 0, 0], done=[false, false, false, false, false]
[PMAT-072] decode step 2: token_ids=[0, 0, 0, 0, 0], done=[false, false, false, false, false]
```

All 5 concurrent requests with different prompts ("Count: 1" through "Count: 5") receive identical garbage output:
```json
{"text":"!!!!"}
```

Prefill works correctly (359 tok/s), but the batched decode step always samples token 0.

## Likely cause

The graph replay/argmax path for batched decode may be reading from the wrong logits buffer, or the batched KV cache indexing is wrong (all sequences reading from the same cache slot).

## Impact

**P0** — GPU batched inference is completely broken. All concurrent requests get the same wrong output.

## Version
`apr 0.4.13` on lambda-labs (RTX 4090)
